src/components/select/select-textbox/select-textbox.spec.js

```diff
@@ -172,38 +172,84 @@ describe("SelectTextbox", () => {
+      const mockAriaLabel = "foo";
+      const wrapper = mount(<SelectTextbox ariaLabel="foo" />);
+      const ariaLabel = wrapper.find(Textbox).prop("aria-label");
+      expect(ariaLabel).toBe(mockAriaLabel);
+    });
+
+    describe.each([
+      ["set", "ariaLabelledBy", "foo"],
+      ["not set", "labelId", undefined],
+    ])(
+      "when the ariaLabelledBy prop is %s",
+      (description, propName, mockAriaLabelledBy) => {
+        const mockLabelId = "baz";
+
+        describe("with the hasTextCursor prop set to true", () => {
+          it(`then the aria-labelledby in the Textbox should contain ${propName} and the id of the SelectText`, () => {
+            const wrapper = mount(
+              <SelectTextbox
+                ariaLabelledBy={mockAriaLabelledBy}
+                hasTextCursor
+                labelId={mockLabelId}
+              />
+            );
+            const ariaLabelledBy = wrapper
+              .find(Textbox)
+              .prop("aria-labelledby");
 
-    describe.each([true, false])(
-      "when labelId is undefined, and hasTextCursor is %s",
-      (hasTextCursor) => {
-        it("aria-labelledby does not point to a non-existent label", () => {
-          const wrapper = mount(
-            <SelectTextbox hasTextCursor={hasTextCursor} labelId={undefined} />
-          );
-          const ariaLabelledBy = wrapper.find(Textbox).prop("aria-labelledby");
-          expect(ariaLabelledBy).not.toEqual(expect.stringContaining(labelId));
+            expect(ariaLabelledBy).toEqual(
+              expect.stringContaining(mockAriaLabelledBy || mockLabelId)
+            );
+            expect(ariaLabelledBy).toEqual(expect.stringContaining(mockTextId));
+          });
+
+          describe("and the accessibilityLabelId prop", () => {
+            it("then the aria-labelledby in the Textbox should contain both id's", () => {
+              const mockAccessibilityLabelId = "bar";
+              const wrapper = mount(
+                <SelectTextbox
+                  accessibilityLabelId={mockAccessibilityLabelId}
+                  ariaLabelledBy={mockAriaLabelledBy}
+                  hasTextCursor
+                  labelId={mockLabelId}
+                />
+              );
+              const ariaLabelledBy = wrapper
+                .find(Textbox)
+                .prop("aria-labelledby");
+
+              expect(ariaLabelledBy).toEqual(
+                expect.stringContaining(mockAriaLabelledBy || mockLabelId)
+              );
+              expect(ariaLabelledBy).toEqual(
+                expect.stringContaining(mockAccessibilityLabelId)
+              );
+            });
+          });
         });
       }
     );
+
+    describe("when the ariaLabelledBy and the ariaLabel prop are set", () => {
+      it("then the ariaLabelledBy prop value should be passed down to the aria-labelledby prop of the Textbox", () => {
```

Comment:
as I explained in the component file, I don't think this behaviour being tested for is quite correct. If you agree with tha tcomment you can just check that the `aria-labelledby` prop contains the `mockAriaLabelledBy` (rather than being equal to it). Or even perhaps remove this test entirely, as both the `aria-label` and `aria-labelledby` attributes should be being covered, separately, by the tests above.

Reply:
this is the expected behaviour,  aria-labelledby in this case should be what user defined, not a combination that is usually passed down from the SelectTextbox, so the attribute should have the same value as the mock, not just contain it.